src/lib/Web3Function.ts

```diff
@@ -63,6 +71,10 @@ export class Web3Function {
               diff: difference,
             },
           },
+          callbacks: {
+            onFail: this._onFail !== undefined,
+            onSuccess: this._onSuccess !== undefined,
+          },
```

Comment:
I think this should be inside `data` 
And it will make processing down the line more uniformed